core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java

```diff
@@ -231,52 +218,36 @@ public void setServiceSelectionStylesheetFilename(String newServiceSelectionStyl
 	public void setXpathExpression(String xpathExpression) {
 		this.xpathExpression = xpathExpression;
 	}
-	public String getXpathExpression() {
-		return xpathExpression;
-	}
 
 	@IbisDoc({"3", "Namespace defintions for xpathExpression. Must be in the form of a comma or space separated list of <code>prefix=namespaceuri</code>-definitions. For some use other cases (NOT xpathExpression), one entry can be without a prefix, that will define the default namespace.", ""})
 	public void setNamespaceDefs(String namespaceDefs) {
 		this.namespaceDefs = namespaceDefs;
 	}
-	public String getNamespaceDefs() {
-		return namespaceDefs;
-	}
 
+	@Deprecated
+	@ConfigurationWarning("Please use 'getInputFromSessionKey' attribute instead")
 	@IbisDoc({"4", "Name of the key in the <code>PipeLineSession</code> to retrieve the input message from, if a styleSheetName or a xpathExpression is specified. " + 
 					"If no styleSheetName or xpathExpression is specified, the value of the session variable is used as the name of the forward. " + 
 					"If none of sessionKey, styleSheetName or xpathExpression are specified, the element name of the root node of the input message is taken as the name of forward.", ""})
 	public void setSessionKey(String sessionKey){
 		this.sessionKey = sessionKey;
-	}
-	public String getSessionKey(){
-		return sessionKey;
+		this.forwardNameSessionKey = sessionKey;
```

Comment:
you cannot assign sessionKey to forwardNameSessionKey, i.m.o.

Reply:
Assignment is for backwards compability. 'sessionKey' attribute could be set by frank developers. Keeping the support for the sessionKey attribute requires either setting the forwardNameSessionKey or having additional if statement in doPipe.

Reply:
Well, you are right actually!